### PR TITLE
Feat(eos_cli_config_gen): BGP VRF IPv4 RM support

### LINKED ISSUE #1756: BGP VRF Address-Family IPv4 Route-map support
### Enhancement summary

please detail your feature enhancement.

We have requirement to apply a route map in Router BGP XX -> vrf xx -> address-family ipv4 level. The route-map command is supported in the EOS shell but we are not able to apply the same route-map in the AVD.

As a work around, we are using the eos_cli to archive the same function. But it would be great to have the route-map variable available for us.

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Below is the required command in EOS command lines:
router bgp 65202
  vrf OFFNET-SVR
    address-family ipv4
      **neighbor x.x.x.x route-map ALI-DC-OFFNET out**

### Describe the solution you would like

Below is the EOS_CLI workaround in the AVD:

custom_structured_configuration_router_bgp:
  vrfs:
    OFFNET-SVR:
      neighbors:
        x.x.x.x:
          remote_as: 65011
      address_families:
        ipv4:
          neighbors:
            x.x.x.x:
              activate: true
      **eos_cli: |
        address-family ipv4
          neighbor 172.27.2.210 route-map ALI-DC-OFFNET out**

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by danieltudares**
Feat(eos_cli_config_gen): BGP VRF IPv4 RM support

## Change Summary

Support AVD data model for BGP VRF address-family IPv4 Route-maps

## Related Issue(s)

Fixes #1756 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Support for knob to apply route maps (in/out) under `router_bgp > vrfs > address_families > neighbor` :

```yaml
router_bgp:
  vrfs:
    <vrf>:
      address_families:
        < address_family >:
          neighbors:
            < neighbor_ip_address >:
              route_map_out: < route-map name >
              route_map_in: < route-map name >
```

## How to test

molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 1]
```diff
@@ -3189,6 +3189,8 @@ router_bgp:
           neighbors:
             < neighbor_ip_address >:
               activate: < true | false >
+              route_map_out: < route-map name >
+              route_map_in: < route-map name >
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]:
```suggestion
              route_map_out: < route_map_name >
              route_map_in: < route_map_name >
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md [hunk 2]
```diff
@@ -2907,6 +2907,8 @@ router_bgp:
           neighbors:
             - ip_address: < neighbor_ip_address >
               activate: < true | false >
+              route_map_out: < route-map name >
+              route_map_in: < route-map name >
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README_v4.0.md]:
```suggestion
              route_map_out: < route_map_name >
              route_map_in: < route_map_name >
```

**Review by gmuloc — COMMENTED** [review]
Just a couple of minor names to change in READMEs to match (most of) the other README indications

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 1 above)
Comment (danieltudares) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: makes sense, thanks!

**Review by gmuloc — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**